{"version":3,"file":"external-proxy.js","sourceRoot":"/Users/hanxiaoxue/Documents/work/frontEndDebugger/nest/packages/core/","sources":["helpers/external-proxy.ts"],"names":[],"mappings":";;;AAEA,8EAAyE;AAEzE,MAAa,kBAAkB;IACtB,WAAW,CAChB,cAAuC,EACvC,iBAA4C,EAC5C,IAAe;QAEf,OAAO,KAAK,EAAE,GAAG,IAAW,EAAE,EAAE;YAC9B,IAAI;gBACF,OAAO,MAAM,cAAc,CAAC,GAAG,IAAI,CAAC,CAAC;aACtC;YAAC,OAAO,CAAC,EAAE;gBACV,MAAM,IAAI,GAAG,IAAI,6CAAoB,CAAC,IAAI,CAAC,CAAC;gBAC5C,IAAI,CAAC,OAAO,CAAW,IAAI,CAAC,CAAC;gBAC7B,OAAO,iBAAiB,CAAC,IAAI,CAAC,CAAC,EAAE,IAAI,CAAC,CAAC;aACxC;QACH,CAAC,CAAC;IACJ,CAAC;CACF;AAhBD,gDAgBC","sourcesContent":["import { ContextType } from '@nestjs/common/interfaces';\nimport { ExternalExceptionsHandler } from '../exceptions/external-exceptions-handler';\nimport { ExecutionContextHost } from '../helpers/execution-context-host';\n\nexport class ExternalErrorProxy {\n  public createProxy<TContext extends string = ContextType>(\n    targetCallback: (...args: any[]) => any,\n    exceptionsHandler: ExternalExceptionsHandler,\n    type?: TContext,\n  ) {\n    return async (...args: any[]) => {\n      try {\n        return await targetCallback(...args);\n      } catch (e) {\n        const host = new ExecutionContextHost(args);\n        host.setType<TContext>(type);\n        return exceptionsHandler.next(e, host);\n      }\n    };\n  }\n}\n"]}